### PR TITLE
created data API endpoint

### PR DESCRIPTION
**Pull request by abhiredd117**
Updated the file to create an endpoint that returns JSON information containing a map of <time, ndvi_data> for cam#, start date, end date

### CODE REVIEW

File: AGICamEndpoint/src/main/java/org/agicam/endpoint/controllers/ImageController.java [hunk 1]
```diff
@@ -13,23 +13,76 @@
...
+        return ndviDataMap;
+    };
+
+    // Method to retrieve NDVI data from the database
+    private static Map<String, Double> getNDVIData(String cameraNumberString, LocalDateTime startDateTime, LocalDateTime endDateTime) {
+        List<Document> ndviDataDocs = ImageColl.get().getCollection()
+                .find(and(
+                        eq("cameraNumber", cameraNumberString),
+                        eq("ndviData", new Document("$exists", true)),
+                        eq("dateTime", new Document("$gte", startDateTime).append("$lte", endDateTime))
```
Comment (jordanzyx) [inline AGICamEndpoint/src/main/java/org/agicam/endpoint/controllers/ImageController.java:54]: This logically does not seem correct. Condition should be
cam# = value 
ndviData exists
gte start && lte end

File: AGICamEndpoint/src/main/java/org/agicam/endpoint/controllers/ImageController.java [hunk 2]
```diff
@@ -13,23 +13,76 @@
...
+        // Set response type to JSON and return NDVI data
+        response.type("application/json");
+        return ndviDataMap;
+    };
+
+    // Method to retrieve NDVI data from the database
+    private static Map<String, Double> getNDVIData(String cameraNumberString, LocalDateTime startDateTime, LocalDateTime endDateTime) {
+        List<Document> ndviDataDocs = ImageColl.get().getCollection()
+                .find(and(
+                        eq("cameraNumber", cameraNumberString),
```
Comment (jordanzyx) [inline AGICamEndpoint/src/main/java/org/agicam/endpoint/controllers/ImageController.java:52]: Check if the value in database is of type string or int. Using the cameraNumberString here could lead to inaccuracies

**Review by jordanzyx — CHANGES_REQUESTED** [review]

File: AGICamEndpoint/src/main/java/org/agicam/endpoint/controllers/ImageController.java [hunk 3]
```diff
@@ -13,23 +13,76 @@
...
+        return ndviDataMap;
+    }
+
+    // Handler for image requests
     public static Route getImage = (Request request, Response response) -> {
         String cameraNumberString = request.queryParams("camNum");
-        String dateTime = request.queryParams("dateTime");
-        String take = request.queryParams("take");
+        String dateTimeString = request.queryParams("dateTime");
+        String takeString = request.queryParams("take");
```
[the remark this answers is not in the document]
Comment (abhiredd117) [reply AGICamEndpoint/src/main/java/org/agicam/endpoint/controllers/ImageController.java:73]: I will change it back

File: AGICamEndpoint/src/main/java/org/agicam/endpoint/controllers/ImageController.java (hunk 1 above)
Comment (abhiredd117) [reply AGICamEndpoint/src/main/java/org/agicam/endpoint/controllers/ImageController.java:54]: Can you elaborate, please 
for cam# = value, 
ndviData exists:
eq("ndviData", new Document("$exists", true)) // value of $exists is set to true, it indicates that the field must exist in the document.
"gte start && lte end:" elaborate for this too

Comment (jordanzyx) [reply AGICamEndpoint/src/main/java/org/agicam/endpoint/controllers/ImageController.java:54]: Maybe I am wrong. Test the code and post results of it working and we can merge.

**Pull request closed without merging**